### ISSUE TITLE
EnumMemeber containing comma in the value field is not deserialized correctly but is serialized correctly

### ISSUE THREAD

**Issue by hovsepm** [Azure/azure-sdk-for-net#4234]
Checked with following runtimes:
https://www.nuget.org/packages/Microsoft.Rest.ClientRuntime.Azure/3.3.12
https://www.nuget.org/packages/Microsoft.Rest.ClientRuntime/2.3.11

The enum is generated like [this](https://github.com/Azure/azure-libraries-for-net/blob/Fluent-v1.8/src/ResourceManagement/Compute/Generated/Models/ResourceIdentityType.cs#L20):
```
    [JsonConverter(typeof(StringEnumConverter))]
    public enum ResourceIdentityType
    {
        [EnumMember(Value = "SystemAssigned")]
        SystemAssigned,
        [EnumMember(Value = "UserAssigned")]
        UserAssigned,
        [EnumMember(Value = "SystemAssigned, UserAssigned")]
        SystemAssignedUserAssigned,
        [EnumMember(Value = "None")]
        None
    }
```
if the value SystemAssignedUserAssigned of the enum is serialized it has correct network payload value of `SystemAssigned, UserAssigned` but when the same value is received from server it is deserialized as `UserAssigned` . 
I'm pretty sure this happens in the autogenerated word as well.

Repro steps in Fluent test:
1. Clone https://github.com/Azure/azure-libraries-for-net/tree/Fluent-v1.8  (tag is important)
2. Open in the root folder Fluent.Tests.sln
3. After build open `\src\ResourceManagement\Compute\Generated\VirtualMachinesOperations.cs` file and put a breakpoint at line 1907.
4. Open test explorer window and start debuging `Fluent.Tests.Compute.VirtualMachine.EMSILMSIOperations.CanCreateWithLMSIAndEMSI` test
5. When your breakpoint will be activated you will see
```
                _responseContent = await _httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
                try
                {
                    _result.Body = Rest.Serialization.SafeJsonConvert.DeserializeObject<VirtualMachineInner>(_responseContent, Client.DeserializationSettings);
                }
```
that _responseContent contains 
```
...
 "identity": {
    "type": "SystemAssigned, UserAssigned",
...
```
but `_result.Body.Identity.Type` will have value `UserAssigned`

**Comment by shahabhijeet** [comment]
Would you mind trying by editing the code and adding the [DataMember] attribute on the ResourceIdentityType and let me know if that makes it deserialize it appropriately.

**Comment by hovsepm** [comment]
Does not help. 
```
    /// <summary>
    /// Defines values for ResourceIdentityType.
    /// </summary>
    [JsonConverter(typeof(StringEnumConverter))]
    [DataContract]
    public enum ResourceIdentityType
    {
        [EnumMember(Value = "SystemAssigned")]
        [DataMember]
        SystemAssigned,
        [EnumMember(Value = "UserAssigned")]
        [DataMember]
        UserAssigned,
        [EnumMember(Value = "SystemAssigned, UserAssigned")]
        [DataMember]
        SystemAssignedUserAssigned,
        [EnumMember(Value = "None")]
        [DataMember]
        None
    }
```

Same issue. BTW `[DataMember] ` is not applicable to the class or enum definition. Only to it's properties/members.

**Comment by shahabhijeet** [comment]
@hovsepm this bug is fixed in newtonsoft 10.0.3
So at this point .NET SDK targeting .NET 4.5.2 will not be fixed, but all SDKs that are targeting .NET Core can upgrade to newtonsoft 10.0.3 or above and this will not be an issue.

.Net SDK targeting .NET 4.5.2 cannot be upgraded due to powershell restriction and hence will have to wait until later part of year 2018
